### ISSUE TITLE
Compile Time check for json decorator

### ISSUE THREAD

**Issue by anyon17** [JairusSW/as-json#93]
Right now a class decorated with `@json` containing fields which might not have types whose declaration have `@json` decorator panics at the runtime. Is there any way to have this check at compile time so that compiler can exactly point to the field which does not have `@json` decorator in it's declaration.

**Comment by anyon17** [comment]
Yeah that's perfectly put @JairusSW . Thanks for attaching the example to make the issue much clear.

**Comment by JairusSW** [comment]
Reopening this as there are a few bugs

**Comment by JairusSW** [comment]
@anyon17, after taking a closer look, it seems like to _properly_ implement this (and a few other things), I'm going to have to take multiple passes over the AST via the transform. I'm not sure if your familiar with transforms, but its a pretty big undertaking, so I'll pull some features out and release a non-feature-complete version that's stable. This means I'll add full type checking in a later release.
I'm still going to be working on this, just thought I'd let you know it may take a bit longer.

**Comment by anyon17** [comment]
hey @JairusSW thanks for letting me know this detail. Will wait for the feature to become stable and looking forward to use it. Thanks again for picking up the requested feature.
